www/js/con-student.js

```diff
@@ -304,12 +290,48 @@ define([
 			// Apply settings from PM
 			console.log("Initial Control Settings for Student");
 			
+			var d = registry.byId(this.controlMap.description);
+			
+			console.log("description widget = ", d, this.controlMap.description);
+			// Add options each time node editor is opened.
+			// By this, we can have quantity and equation options for those nodes
+			d.removeOption(d.getOptions()); // Delete all options
+
+			//get descriptions to sort as alphabetic order
+			var descriptions = this._model.authored.getDescriptions();
+			//create map of names for sorting
+			var descNameMap = {};
+			array.forEach(descriptions, function (desc) {
+				var _name = this._model.authored.getName(desc.value);
+				descNameMap[desc.value] = _name;
+			}, this);
+			descriptions.sort(function (obj1, obj2) {
+				//return obj1.label > obj2.label;
+				if(descNameMap[obj1.value] && descNameMap[obj2.value])
+					return descNameMap[obj1.value].toLowerCase().localeCompare(descNameMap[obj2.value].toLowerCase());
```

Comment:
this code should be in model.js as well. The controller should have code to render the field data. The field data manipulation should be part of the model. Like we getAllUnits, the idea is something similar.

Reply:
@dharanisakary I see all changes Sachin requested have been made except this one.  Was there a reason this one is unchanged?

Reply:
These changes aren't a part of this pull request. Also, I added new function to model.js to sort the descriptions by variable name. So, after review of those changes, I think we can use that function here somehow. Please let me know your views on this.

Reply:
@sachingrover211 Dharani has suggested we wait and make this change after #241.  If that's ok I'll merge this PR and add a note to #241 to do it there.